File: 09.Greed贪心/leetcode_2091_minimumDeletions.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    int minimumDeletions(vector<int>& nums) {
        int mx = *max_element(nums.begin(), nums.end());
        int mn = *min_element(nums.begin(), nums.end());
        int n = nums.size();
        int p1 = 0, p2 = 0;
        for (int i=0; i<nums.size(); i++){
            if (nums[i] == mx){
                p1 = i;
            }
            if (nums[i] == mn){
                p2 = i;
            }
        }
        if (p1 > p2){
            swap(p1, p2);
        }
        return n-max(p1, max(n-p2-1, p2-p1-1));
    }
};

int main(){
    Solution s;
    vector<int> nums = {2,10,7,5,4,1,8,6};
    cout << s.minimumDeletions(nums) << endl;
}

File: 09.Greed贪心/leetcode_2091_minimumDeletions.py
from typing import List

class Solution:
    def minimumDeletions(self, nums: List[int]) -> int:
        mx, mn = max(nums), min(nums)
        nums.append(mx)
        n = len(nums)
        last = -1
        res = n-1
        for i in range(len(nums)):
            v = nums[i]
            if v==mx or v==mn:
                res = min(res, n-1-(i-last-1))
                last = i
        return res

if __name__ == "__main__":
    s = Solution()
    nums = [2,10,7,5,4,1,8,6]
    print(s.minimumDeletions(nums))
